### PR TITLE
Adds micronaut example.

### PR DESCRIPTION
**Pull request by coollog**
Part of #540 
Copied over from https://github.com/coollog/micronaut-jib

Preview at https://github.com/GoogleContainerTools/jib/tree/micronaut-example/examples/micronaut

### CODE REVIEW

File: examples/micronaut/src/main/groovy/example/micronaut/HelloController.groovy [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import groovy.transform.CompileStatic
+import io.micronaut.http.annotation.Controller
+import io.micronaut.http.annotation.Get
+
+@CompileStatic
+@Controller("/hello") // <1>
+class HelloController {
+    @Get("/") // <2>
+    String index() {
+        "Hello Jib" // <3>
```
Comment (TadCordle) [inline examples/micronaut/src/main/groovy/example/micronaut/HelloController.groovy]: Everything on the README says the output is "Hello World", maybe this should match that.

Comment (coollog) [reply examples/micronaut/src/main/groovy/example/micronaut/HelloController.groovy]: Fixed

File: examples/micronaut/README.md [hunk 2]
```diff
@@ -20,7 +20,7 @@ docker run -d -p 8080:8080 micronaut-jib:0.1
 ```
 ```shell
 curl localhost:8080/hello
-> Hello World
+> Hello Jib
```
Comment (TadCordle) [inline examples/micronaut/README.md]: Maybe we should change the code instead, the gif still says Hello World.

Comment (coollog) [reply examples/micronaut/README.md]: Ah, good point, I'll just change back to Hello World

File: examples/micronaut/build.gradle [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        exclude group: 'org.codehaus.groovy', module: 'groovy-all'
+    }
+    testCompile 'io.micronaut:inject-java'
+}
+
+compileJava.options.compilerArgs += '-parameters'
+compileTestJava.options.compilerArgs += '-parameters'
+
+/** CHANGE THIS TO THE IMAGE YOU WANT TO PUSH TO */
+// jib.to.image = 'gcr.io/PROJECT_ID/micronaut-jib'
```
[the remark this answers is not in the document]
Comment (coollog) [reply examples/micronaut/build.gradle]: Yea this line is just there to give the user an example of how to configure `jib-gradle-plugin` if they want to build directly to a registry. The project version is 0.1 so you would need to use the `micronaut-jib:0.1` reference that Jib generates for `jibDockerBuild`.

[the remark this answers is not in the document]
Comment (chanseokoh) [reply examples/micronaut/build.gradle]: Ah, that was my copy-paste error. The only thing to fix is to add the `--image=micronaut-jib` parameter then.

https://github.com/GoogleContainerTools/jib/pull/893/files#diff-0a4703c3cc2535016cc80dff59feaa82R17

[the remark this answers is not in the document]
Comment (coollog) [reply examples/micronaut/build.gradle]: Yep, the reference is generated by `jibDockerBuild`

**Pull request merged**